utils/src/lib.rs

```diff
@@ -505,6 +505,28 @@ pub async fn fourbyte_event(topic: &str) -> Result<Vec<(String, i32)>> {
         .collect::<Vec<(String, i32)>>())
 }
 
+pub async fn pretty_calldata(calldata: &str) -> Result<u8> {
+    let sig = calldata.get(..10).expect("size cannot be less than 4 bytes");
+    let selector = fourbyte_possible_sigs(sig, Some(String::from("earliest"))).await?;
+    let data = calldata.get(10..).unwrap();
+
+    if data.len() % 64 != 0 {
+        panic!("Invalid calldata size")
+    }
+
+    match selector.len() {
```

Comment:
I guess this can be if selector.is_empty() {} else {}
instead?

Reply:
Yess, will make the changes, wasn't aware of that method :)